src/pages/main.test.tsx

```diff
@@ -13,6 +46,113 @@ describe('Main page', () => {
     expect(screen.getByTestId('header')).toBeInTheDocument();
     expect(screen.queryAllByRole('link').length).toBeGreaterThan(0);
     expect(screen.getByTestId('search-bar')).toBeInTheDocument();
-    expect(screen.getByTestId('items-cards-list')).toBeInTheDocument();
+    expect(screen.getByAltText('Loading...')).toBeInTheDocument();
+    expect(await screen.findByTestId('items-cards-list')).toBeInTheDocument();
+    expect(screen.queryByAltText('Loading...')).not.toBeInTheDocument();
+    expect(screen.getAllByTestId('item-card').length).toBe(itemsData.length);
+    expect(screen.queryByTestId('err-msg')).not.toBeInTheDocument();
+    expect(fakeFetch).lastCalledWith(
+      'https://the-one-api.dev/v2/character/?page=1&limit=10&name=//i',
+      options
+    );
+  });
+
+  it('submits search with results by Enter key pressing in SearchBar', async () => {
+    render(
+      <BrowserRouter>
+        <Main />
+      </BrowserRouter>
+    );
+    expect(await screen.findByTestId('items-cards-list')).toBeInTheDocument();
+    userEvent.type(screen.getByRole('textbox'), '{selectall}Search with result{Enter}');
+    expect(await screen.findByAltText('Loading...')).toBeInTheDocument();
+    expect(await screen.findByTestId('items-cards-list')).toBeInTheDocument();
+    expect(screen.queryByAltText('Loading...')).not.toBeInTheDocument();
+    expect(screen.getAllByTestId('item-card').length).toBe(itemsData.length);
+    expect(screen.queryByTestId('err-msg')).not.toBeInTheDocument();
+    expect(fakeFetch).toBeCalledWith(
+      'https://the-one-api.dev/v2/character/?page=1&limit=10&name=/Search with result/i',
+      options
+    );
+  });
+
+  it('submits search without results by Search button click', async () => {
+    render(
+      <BrowserRouter>
+        <Main />
+      </BrowserRouter>
+    );
+    expect(await screen.findByTestId('items-cards-list')).toBeInTheDocument();
+
+    fakeFetch = jest.spyOn(global, 'fetch').mockImplementation(async () => {
+      return new Promise(async (resolve) => {
+        const sleep = async (delay: number) => setTimeout(() => {}, delay);
```

Comment:
Actually, it is the incorrect implementation of sleep. You should put `resolve` into setTimeout's callback to make it actually wait for 1 second